doc/user/content/transform-data/troubleshooting.md

```diff
@@ -93,11 +95,35 @@ this means that the index was correctly used. If that's not the case, consider:
 * Does the index's indexed expression (key) match up with how you're querying
   the data?
 
-#### Computation-intensive queries
-
-Another thing to be aware of, in Materialize, aggregations like `COUNT(*)` and limit constraints that return more than 25 rows like `LIMIT 50` perform a computation across the full underlying dataset being queried, so can be resource intensive. If you want to execute these types of queries you should either:
-* Create a materialized view or index for these queries if you plan to issue them often
-* Use a large enough cluster to fit the full underlying dataset being queried
+#### Resource-intensive queries
+##### Data-intensive queries
+Queries with aggregations, ordering, or offsets can be data-intensive, because
+they need to read in the full underlying dataset in order to compute the
+result. Some examples are operations like `COUNT()`, `MAX()` and `ORDER BY`.
+If the underlying data is not indexed, then the query execution needs to read
+it in from storage. Ingesting all that data can take time, causing your query
+to run slowly.
+
+##### Computation-intensive queries
+Other queries, like those with `JOIN`s can be computation-intensive.
+Performing those computations on the fly can also take time. During the execution
+of the query, Materialize will temporarily use memory to store the joined
+data and CPU to execute the join. You can read more about the impacts of CPU
```

Comment:
This isn't right if the data are already indexed. It will take some amount of *time*, but it will not additionally use memory to store the join inputs.